homeassistant/components/n26/__init__.py

```diff
@@ -0,0 +1,93 @@
+"""Support for N26 bank accounts.
+
+https://n26.com
+
+N26 is a bank from germany that launched as a start up.
+It is an "online only" bank in that it has no local "stores"
+to retrieve or deposit money and account management is done only
+through their web interface and mobile apps.
+"""
+from datetime import datetime, timedelta, timezone
+from functools import partial
+import logging
+
+import voluptuous as vol
+
+from homeassistant.const import (
+    CONF_PASSWORD, CONF_SCAN_INTERVAL, CONF_USERNAME)
+from homeassistant.helpers import discovery
+import homeassistant.helpers.config_validation as cv
+
+REQUIREMENTS = ['n26==0.2.2']
+
+_LOGGER = logging.getLogger(__name__)
+
+DOMAIN = 'n26'
+
+CONF_CARD_ID = "card_id"
+DEFAULT_SCAN_INTERVAL = timedelta(seconds=1800)
+
+# define configuration parameters
+CONFIG_SCHEMA = vol.Schema({
+    DOMAIN: vol.Schema({
+        vol.Required(CONF_USERNAME): cv.string,
+        vol.Required(CONF_PASSWORD): cv.string,
+        vol.Required(CONF_CARD_ID): cv.string,
+        vol.Optional(CONF_SCAN_INTERVAL,
+                     default=DEFAULT_SCAN_INTERVAL): cv.time_period,
+    }),
+}, extra=vol.ALLOW_EXTRA)
+
+N26_COMPONENTS = [
+    'sensor',
+    'switch'
+]
+
+API = "api"
+ACCOUNT_INFO = "account_info"
+CARDS = "cards"
+SPACES = "spaces"
+
+
+async def async_setup(hass, config):
+    """Set up N26 Component."""
+    _LOGGER.debug("Initializing N26")
+
+    user = config[DOMAIN][CONF_USERNAME]
+    password = config[DOMAIN][CONF_PASSWORD]
+    card_id = config[DOMAIN][CONF_CARD_ID]
+
+    # initialize N26 API
+    from n26 import api, config as api_config
+    n26_api = api.Api(api_config.Config(user, password, card_id))
+
+    hass.data[DOMAIN] = {}
+
+    account_info = await hass.async_add_executor_job(
+        partial(n26_api.get_account_info))
```

Comment:
why  `partial`?

Reply:
good catch, I copied it from another component I created for hass which had arguments.
But I guess it's better to remove the asyncio code from this component entirely if the API doesn't support it (and I doubt it ever will).